pkg/gofr/datasource/kv-store/badger/badger.go

```diff
@@ -0,0 +1,103 @@
+package badger
+
+import (
+	"context"
+	"github.com/dgraph-io/badger/v4"
+)
+
+type Configs struct {
+	DirPath string
+}
+
+type client struct {
+	db      *badger.DB
+	configs Configs
+	logger  Logger
+	metrics Metrics
+}
+
+func New(configs Configs) *client {
+	return &client{configs: configs}
+}
+
+// UseLogger sets the logger for the MongoDB client which asserts the Logger interface.
+func (c *client) UseLogger(logger interface{}) {
+	if l, ok := logger.(Logger); ok {
+		c.logger = l
+	}
+}
+
+// UseMetrics sets the metrics for the MongoDB client which asserts the Metrics interface.
+func (c *client) UseMetrics(metrics interface{}) {
```

Comment:
Same question about using


```suggestion
func (c *client) UseMetrics(metrics Metrics) {
```

Reply:
provider interface has interface, that is why we are accepting interfaces.
Else we would need to have a common source for the Logger and Metrics interface which would bring dependency, which is we are refraining from